Assets/Scripts/Monster.cs

```diff
@@ -1,134 +1,42 @@
     }
 
-    private void Start()
-    {
-        _newtonHealth = startingHealth;
-    }
-
-    private void FixedUpdate()
+    private new void OnCollisionEnter2D(Collision2D collision)
     {
-        
-    }
-
-    private void OnCollisionEnter2D(Collision2D collision)
-    {
-        CollisionHandler(collision);
-    }
-
-    private void CollisionHandler(Collision2D collision)
-    {
-        if (isDead) return;
-
-        // if collision is with a bird, always consider it a kill
-        Bird bird = collision.gameObject.GetComponent<Bird>();
-        if (bird)
+        if (isDead)
         {
-            Die();
+            return;
         }
-
-        // monsters are squishy and won't hurt each other
-        Monster anotherMonster = collision.gameObject.GetComponent<Monster>();
-        if (anotherMonster) return;
-
-        // if something falls onto the monster
-        //if (collision.contacts[0].normal.y < -0.5)
-
-        float impactDamage = 0;
-
-        if(collision.gameObject.tag == "BluntObject")
+        else
         {
-            var otherAccelerometer = collision.gameObject.GetComponent<Accelerometer>();
-
-            var relativeVelocity = accelerometer.velocity - otherAccelerometer.velocity;
-
-            if (accelerometer.velocity.magnitude > almostZero && otherAccelerometer.velocity.magnitude > almostZero)
-            {
-                // both objects moving
-                impactDamage = CalculateImpactDamage((rigidbody.mass + collision.rigidbody.mass), relativeVelocity, ouchForce);
-            }
-            else if(accelerometer.velocity.magnitude > almostZero)
-            {
-                // monster is moving
-                impactDamage = CalculateImpactDamage(rigidbody.mass, relativeVelocity, ouchForce);
-            }
-            else if(otherAccelerometer.velocity.magnitude > almostZero)
-            {
-                // other is moving
-                impactDamage = CalculateImpactDamage(collision.rigidbody.mass, relativeVelocity, ouchForce);
-            }
+            base.OnCollisionEnter2D(collision);
         }
-
-        if (impactDamage > 0) ApplyDamage(impactDamage);
-    }
-
-    private float CalculateImpactDamage(float mass, Vector2 relativeVelocity, float minForce)
-    {
-        var exertedForce = 0.5f * mass * Mathf.Pow(relativeVelocity.magnitude, 2);
-
-        if (exertedForce > minForce) return exertedForce;
-        else return 0;
-    }
-
-    private void ApplyDamage(float newtonDamage)
-    {
-        _newtonHealth -= newtonDamage;
-        if (_newtonHealth <= 0) Die();
+        
     }
 
-    private void Die()
+    public override void DeathRoutine()
     {
-        // prevent permanent negative numbers
-        _newtonHealth = 0;
-
         spriteRenderer.sprite = deadSprite;
         poofParticleSystem.Play();
-
-        StartCoroutine(DeactivateAfterDelay());
```

Comment:
Based on my comment above with the DeactivateAfterDelay change, can we pass in the "animation" time from the poofParticleSystem instead of a hard coded 1.5f?

Reply:
Done: https://github.com/cjh15hub/terribletweeters/commit/81f82db7e00559acff3871b77076b9bdb9bcaa43#diff-021dc1e9eb0584e8b459501c5c0ad3d7dfab210bd7cd63a72fb3ee576efbb985R26